internal/pkg/pkg.go

```diff
@@ -146,6 +150,26 @@ func New(path string) (*Pkg, error) {
 	return pkg, nil
 }
 
+// NewWithFs returns a pkg given an absolute or relative OS-defined path.
+// It uses the input FileSystem implementation as opposed to FileSystem on disk
+func NewWithFs(path string, fs filesys.FileSystem) (*Pkg, error) {
```

Comment:
Would suggest merging this with `New` and making fsys mandatory. We can move the `os.GetCwd` call to the caller of the `Render()`.

Reply:
:D @droot  I just asked a similar question the other way around - if there could be two forms where the old version calls the new version until it's not needed. Also if the paths could change from an absolute path to fs.FS plus relative path.

The concern being exactly what you mentioned about avoiding mix-mode conventions. With the default FileSystemOrDisk being passed as the default from so many place, and the Pkg still knowing an absolute path - it seems it would be very easy to have parts of the code that "de-virtualize" the FS.

Especially tricky because those bugs will be invisible as while you're running/testing os dir paths.

Reply:
That said - assuming we want to change the signature in-place instead of having a temporary doubling up of functions - instead of shifting from `absPath string` to `fsys filesys.FileSystem, absPath string` --- should we go to `fsys fs.FS, relPath string`?

Reply:
It is better to go with absolute paths as Sunil mentioned. fs.FS is particular implementation of file system and I wanted to keep the input abstract. fs.FS only has read methods. Users can build their own FileSystem extending fs.FS and pass it. We can iterate on this if we see use-cases in the future.